external-api/src/repositories/estore/estore-repo.ts

```diff
@@ -0,0 +1,86 @@
+import { InsertOneResult, ObjectId, UpdateFilter, UpdateResult, WithId, WithoutId } from 'mongodb';
+import { MONGO_DB_COLLECTIONS } from '@ukef/dtfs2-common';
+import { getCollection } from '../../database';
+
+/**
+ * Finds a document in the CRON_JOB_LOGS collection based on the provided deal ID.
+ *
+ * @param {ObjectId} dealId - The unique identifier for the deal.
+ *
+ * @returns {Promise<WithId<ObjectId> | WithoutId<unknown> | null>} - A promise that resolves to the found document, or `null` if no document is found.
+ *
+ * @throws {Error} - Throws an error if the find operation fails.
+ *
+ * @example
+ * const dealId = new ObjectId('507f1f77bcf86cd799439011');
+ *
+ * find(dealId)
+ *   .then((document) => console.log('Document found', document))
+ *   .catch((error) => console.error('Find operation failed', error));
+ */
+export const find = async (dealId: ObjectId): Promise<WithId<ObjectId> | WithoutId<unknown> | null> => {
```

Comment:
Could this be `findByDealId` to make it specific? I'd expect a standard `find` to accept the same input parameters as the `mongodb` `Collection.find` method, so would be good to make it specific